pkg/workloads/console/controller.go

```diff
@@ -121,22 +126,44 @@ func (r *ConsoleReconciler) Reconcile(request reconcile.Request) (res reconcile.
 			return res, err
 		}
 
-		logger.Log("event", EventNotFound, "resource", Job)
-		job = buildJob(name, consoleTemplate.Spec.Template)
-		if err := controllerutil.SetControllerReference(csl, job, scheme.Scheme); err != nil {
-			return res, err
-		}
-		if err = r.client.Create(r.ctx, job); err != nil {
-			return res, err
+		// Only create a console if it hasn't already expired (and therefore been
+		// deleted)
+		if csl.Status.ExpiryTime == nil {
+			job = buildJob(request.NamespacedName, consoleTemplate.Spec.Template)
+			err = r.client.Create(r.ctx, job)
+			if err != nil {
+				return res, err
+			}
+			logger.Log(
+				"event", EventCreated,
+				"resource", Job,
+				"name", name.Name,
+				"user", csl.Spec.User,
+			)
+
+			// Now that we've spawned a console job we can update the status, as
+			// the expiry and phase will have changed.
+			csl, err = r.updateStatus(csl, job)
```

Comment:
I wonder if it'd be simpler if we `defer` this so it always runs at the end of reconciliation?

Reply:
Not sure if this is possible, as you want to to change the return parameters based on the result of the `updateStatus` call (e.g. if updating the status fails then you want to fail, so that the reconcile gets requeued)

AFAIK you can't change return parameters in a defer func? Maybe there's some pattern that you could apply to achieve this, but unless it's super clean then I'm not sure it's worth it, as I reckon there's only going to be 2 `updateStatus` calls in the reconcile loop. WDYT?

Reply:
Ah yeah true. Probably won't work, then

Reply:
You can change return parameters in the defer! Just assigning to a named returned variable will change it for the return of the function.